azurelinuxagent/ga/exthandlers.py

```diff
@@ -310,11 +311,74 @@ def handle_ext_handlers(self, etag=None):
                 logger.info("Extension handling is on hold")
                 return
 
+        wait_until = datetime.datetime.utcnow() + datetime.timedelta(minutes=DEFAULT_EXT_TIMEOUT_MINUTES)
+        prev_handler = None
+        ext_seq_required = self.is_ext_seq_required()
+
         self.ext_handlers.extHandlers.sort(key=operator.methodcaller('sort_key'))
         for ext_handler in self.ext_handlers.extHandlers:
-            # TODO: handle install in sequence, enable in parallel
+            if ext_seq_required:
+                # Make sure that the previous extension is handled.
+                # If handled successfully, proceed with the current handler.
+                # Otherwise, skip the rest of them.
+                dep_level = ext_handler.sort_key()
+                if dep_level >= 0:
```

Comment:
Let's add a comment to explain the >=0 (we do this only for install/enable and not for uninstall/disable)